python_scripts/functions.py

```diff
@@ -269,11 +263,48 @@ def local_file_to_s3(local_path, s3_path):
     with open(local_path, "rb") as f:
         s3_client.upload_fileobj(f, b, o)
 
+
+
 def generate_iam_config(config, outpath="iam_config.yaml"):
     """
     Should take the necessary paths in the config and write out an
     iam that has access to those s3 paths e.g.
     write-only to log-base-path
     read/write to land-base-path
     """
-    pass
\ No newline at end of file
+
+    expected_keys = [
+        "iam-role-name",
+        "log-base-path",
+        "land-base-path",
+        "pass-base-path",
+        "fail-base-path"
+    ]
+
+    missing_keys = [key for key in expected_keys if key not in config]
+
+    if len(missing_keys) == 0:
+        out_iam = {
+            "iam-role-name": config["iam-role-name"],
+            "athena": {
+                "write": True
+            },
+            "s3": {
+                "write_only": [
+                    os.path.join(config['log-base-path'],'*')
+                ],
+                "read_write": [
+                    os.path.join(config['land-base-path'],'*'),
+                    os.path.join(config['fail-base-path'],'*'),
+                    os.path.join(config['pass-base-path'],'*'),
```

Comment:
We might want to consider making the way we write the config paths the same as `iam_builder` which I think already has the `/*` at the end. Could be wrong though